firewood/src/merkle.rs

```diff
@@ -371,13 +367,23 @@ impl<S: ShaleStore<Node> + Send + Sync> Merkle<S> {
         // have to do some splitting
         let mut val = Some(val);
 
+        // we use Nibbles::<1> so that 1 zero nibble is at the front
+        // this is for the sentinel node, which avoids moving the root
+        // and always only has one child
+        let mut key_nibbles = Nibbles::<1>::new(key.as_ref()).into_iter();
+
         // walk down the merkle tree starting from next_node, currently the root
-        for (key_nib_offset, key_nib) in key_nibbles.into_iter().enumerate() {
+        loop {
+            let Some(key_nib) = key_nibbles.next() else {
+                break;
+            };
+
             // special handling for extension nodes
             if nskip > 0 {
```

Comment:
nit: this could be optimized with `iter::skip(nskip)`, maybe even tossing `nskip` and just doing it when it's set

Reply:
haha I'm in the middle of tossing it for a follow up